### ISSUE TITLE
Error on warp with Industrial Revolution Warptorio Compatibility

### ISSUE THREAD

**Issue by SarumCatt** [Deadlock989/IndustrialRevolution#447]
The mod Industrial Revolution Warptorio Compatibility (0.1.0) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event ir-warptorio-compat::Custom event (ID 234)
No such function: ir-world.allow-surface-fissures
stack traceback:
[C]: in function 'call'
ir-warptorio-compat/control.lua:22: in function 'add_surface_modifications'
ir-warptorio-compat/control.lua:7: in function <ir-warptorio-compat/control.lua:3>
stack traceback:
[C]: in function 'raise_event'
warptorio2/lib/lib_control.lua:258: in function 'vraise'
warptorio2/control_main.lua:858: in function 'Warpout'
warptorio2/control_main_helpers.lua:227: in function 'b'
warptorio2/lib/lib_control.lua:281: in function <warptorio2/lib/lib_control.lua:281>

Modset was Industrial Revolution Warptorio Compatibility and dependencies (latest updates of the following mods: Warptorio2, IR3 + its assets, Planetorio)

Checked for different mod versions. This error starts appearing with Industrial Revolution ver. 3.1.24, so the reason should be starting area generation.

My opinion on the solution is to try and carry out new starting area generation as a mod option to be disabled.

**Comment by Deadlock989** [comment]
Industrial Revolution Warptorio Compatibility is a third party mod.

The error is happening because IRWC is trying to use a remote call that was removed from IR3. The remote call was originally flagged up in the changelog as deprecated in version 3.1.17 (January 2024) with a warning that it no longer had any effect and would be removed at some point in the future. It was eventually removed around June, six months later. I think six months is fair warning. This all happened because various world generation features were switched from scripted events to properly specified autoplace systems, which is why fissures have their own sliders in the world generator options now.

In general it is not ideal to assume that remote calls exist, because if they don't, you will get exactly the kind of error above. IRWC could avoid the error by testing to see if IR3 provides such a call before trying to execute it.

There is nothing I can do to improve this issue. Even if I added new options, the oversight still needs correcting in the third party mod.
